### PR TITLE
Add xml support to HTTP monitors

### PR DESCRIPTION
**Pull request by bayramberkay**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

There is an existing pr request but I guess it hasn't been active for a long time. That's why I just wanted to make a pr request by copy-pasting the xml part. I think it will help most people waiting for this feature.

Fixes #(issue)

https://github.com/louislam/uptime-kuma/issues/1679
https://github.com/louislam/uptime-kuma/issues/1332
https://github.com/louislam/uptime-kuma/pull/1987
https://github.com/louislam/uptime-kuma/pull/2497

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

209863902-a0b37350-d1c8-4e15-bbbe-fa9e7f2c3990

### CODE REVIEW

**Comment by bayramberkay** [comment]
@louislam ,  @JustinTisdale commits are protected as we speak.  Can we go live?

**Comment by chakflying** [comment]
Should users be able to override the contentType by setting custom headers?

**Comment by louislam** [comment]
> Should users be able to override the contentType by setting custom headers?

Custom headers should be the higher priority.

**Comment by Genc** [comment]
> > Should users be able to override the contentType by setting custom headers?
> 
> Custom headers should be the higher priority.

How should we continue? Is the current state enough?

**Comment by Genc** [comment]
How can we advance the process this pr? @louislam  @chakflying

File: server/model/monitor.js [hunk 1]
```diff
@@ -270,17 +271,29 @@ class Monitor extends BeanModel {
...
                         method: (this.method || "get").toLowerCase(),
-                        ...(this.body ? { data: JSON.parse(this.body) } : {}),
+                        ...(bodyValue ? { data: bodyValue } : {}),
                         timeout: this.interval * 1000 * 0.8,
                         headers: {
                             "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
                             "User-Agent": "Uptime-Kuma/" + version,
                             ...(this.headers ? JSON.parse(this.headers) : {}),
                             ...(basicAuthHeader),
+                            ...(contentType ? { "Content-Type": contentType } : {})
```
Comment (chakflying) [inline server/model/monitor.js:296]:
```suggestion
                            ...(contentType ? { "Content-Type": contentType } : {}),
                            ...(basicAuthHeader),
                            ...(this.headers ? JSON.parse(this.headers) : {})
```

**Review by chakflying — CHANGES_REQUESTED** [review]

File: server/model/monitor.js (hunk 1 above)
Comment (Genc) [reply server/model/monitor.js]: Hi @chakflying  , I fix it with this commit : https://github.com/louislam/uptime-kuma/pull/2610/commits/aef85078eb5fbb0e48db31cd40f5d7eaf6c06792

**Comment by Genc** [comment]
How should we continue the process as there are some friends waiting for this? @louislam @chakflying

File: db/patch-http-body-encoding.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+-- You should not modify if this have pushed to Github, unless it does serious wrong with the db.
+BEGIN TRANSACTION;
+
+ALTER TABLE [monitor] ADD http_body_encoding TEXT;
```
Comment (louislam) [inline db/patch-http-body-encoding.sql:4]: VARCHAR(20) should be enough?

Comment (Genc) [reply db/patch-http-body-encoding.sql]: Fixed. But I configure it to 25 characters because "x-www-form-urlencoded" can be added here in the future.

**Comment by louislam** [comment]
Found 2 issues:

1. Body Encoding will be empty for old monitors
image

2. The example placeholder is still in JSON format.
image

**Comment by Genc** [comment]
> Found 2 issues:
> 
> 1. Body Encoding will be empty for old monitors
>    image
> 2. The example placeholder is still in JSON format.
>    image

Thank you very much for your attention. I have good news, fixed them!

**Comment by danwie** [comment]
@Genc Could you please explain why Body Encoding: "Form" was removed in #2610 ? How are we supposed to make a ordinary POST (x-www-form-urlencoded) when the only options are Json/XML?

**Comment by anderskvist** [comment]
I have created this PR that should be ready to merge that will add x-www-form-urlencoded requests - https://github.com/louislam/uptime-kuma/pull/3499

**Pull request merged**